### ISSUE TITLE
Linux kernel ephemeral port range ignored in Docker 20+

### ISSUE THREAD

**Issue by jamiegDS** [moby/moby#43054]
**Description**

According to the "publish" (-P) section of Docker run docs (https://docs.docker.com/engine/reference/run/):
"Docker binds each exposed port to a random port on the host. The range of ports are within an ephemeral port range defined by /proc/sys/net/ipv4/ip_local_port_range."

In Docker versions pre-20, this was correct and working in our environment. However, since upgrading to Docker versions 20+, the kernel ip_local_port_range value is ignored, and Docker instead assigns a port from 49153 through 65535.  This actually affects both the -P option where one publishes all exposed ports, and the -p option where one publishes a single port **and** does not provide a host port. 

I have tested this on Ubuntu Focal, and Flatcar and both exhibit the same behaviour. 

**Steps to reproduce the issue:**
1. Set ip_local_port_range value `sudo sysctl -w net.ipv4.ip_local_port_range="32678 40000"`
2. Make sure docker is stopped and then install docker-ce/docker-ce-cli packages, any 19.X version
3. Start a test container publishing all exposed ports or with a single published port, without providing a host port/host port range. (eg ` docker run -d -p :80 nginx`or `docker run -d -P nginx`)
4. Check the host port assigned with `docker ps` or `docker port`. It will be within 32678-40000 range.
5. Remove container. Stop docker
6. Install docker-ce/docker-ce-cli packages, any 20.X version. Start docker
7. Start a test container publishing all exposed ports or with a single published port, without providing a host port/host port range. (eg. ` docker run -d -p :80 nginx` or `docker run -d -P nginx`)
8. Check the host port assigned with `docker ps` or `docker port`. It will be within 49153-65535 range.

**Describe the results you received:**

A port is assigned from 49153-65535 ephemeral range

**Describe the results you expected:**

A port is assigned from the ephemeral range set by the kernel parameter `/proc/sys/net/ipv4/ip_local_port_range`

**Additional information you deem important (e.g. issue happens only occasionally):**

Our use case here is, we are running many containers on AWS ECS. We would like to control the ports assigned to containers so we can specifically allow these ports through security groups. This is more desirable for us than allowing the whole ephemeral range 32678-65535.

**Output of `docker version`:**

Tested on Ubuntu Focal (I provided minimum 20.10 version, but 20.10.11 also affected)

```
Client: Docker Engine - Community
 Version:           20.10.0
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        7287ab3
 Built:             Tue Dec  8 18:59:40 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.0
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       eeddea2
  Built:            Tue Dec  8 18:57:45 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.4.2-docker)
  scan: Docker Scan (Docker Inc., v0.9.0)

Server:
 Containers: 24
  Running: 3
  Paused: 0
  Stopped: 21
 Images: 42
 Server Version: 20.10.0
 Storage Driver: zfs
  Zpool: xpool
  Zpool Health: ONLINE
  Parent Dataset: xpool/docker
  Space Used By Parent: 3931974144
  Space Available: 131577568768
  Parent Quota: no
  Compression: lz4
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc version: v1.0.2-0-g52b36a2
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.11.0-27-generic
 Operating System: Ubuntu 20.04.3 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 22.97GiB
 Name: ds0377
 ID: 4LGQ:IJGY:AFAY:XS6R:AD6D:JRIE:QX42:D5CD:U74W:34MN:VA7E:G5PV
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 40
  Goroutines: 46
  System Time: 2021-12-01T13:49:21.851318574Z
  EventsListeners: 0
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio weight support
WARNING: No blkio weight_device support

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Both AWS ECS (running Flatcar 2983.2.1/docker - 20.10.11) and physical laptop Ubuntu Focal.

**Comment by thaJeztah** [comment]
hmm.. interesting. Code for that (as far as I'm aware) didn't change.

This bit should be reading the ephemeral port range; https://github.com/moby/moby/blob/5e62ca1a05f2aab49066d91c9070cb7dc7ad2461/libnetwork/portallocator/portallocator_linux.go#L9-L25

From this code, I see a warning / info message should be printed if it falls back to the default port-range (in case it fails to get that information); https://github.com/moby/moby/blob/5e62ca1a05f2aab49066d91c9070cb7dc7ad2461/libnetwork/portallocator/portallocator.go#L107-L117

Do you see a log message like that in the daemon logs? ("falling back to default port range ..")

Just to be sure; are you able to post the output of `cat /proc/sys/net/ipv4/ip_local_port_range` ? (I really don't expect kernel changes would change that format though, as that would be a breaking change)

**Comment by thaJeztah** [comment]
oh, actually, there may have been some changes after the 19.03 release; https://github.com/moby/libnetwork/pull/2461 (not sure if those could cause this)

**Comment by jamiegDS** [comment]
Please see output:
```jgibbard-~/.../ecs$ docker version
Client: Docker Engine - Community
 Version:           20.10.11
 API version:       1.41
 Go version:        go1.16.9
 Git commit:        dea9396
 Built:             Thu Nov 18 00:37:06 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.11
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.9
  Git commit:       847da18
  Built:            Thu Nov 18 00:35:15 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
jgibbard-~/.../ecs$ cat /proc/sys/net/ipv4/ip_local_port_range
32678	40000
jgibbard-~/.../ecs$ docker run --name testinghp -d -p :80 testing123
38a9f7287ecd343b3de974f93819727bdcb22a5b43a8ea2949928e0fc2db613d
jgibbard-~/.../ecs$ docker port 38a9f7287ecd343b3de974f93819727bdcb22a5b43a8ea2949928e0fc2db613d
80/tcp -> 0.0.0.0:49153
80/tcp -> :::49153
```

I do not see any message similar to `falling back to default port range...`, even in debug mode. 

I don't see anything useful but here is a paste of my full journal at the time I started a container:

```
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.124812080Z" level=debug msg="Calling HEAD /_ping"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.125586940Z" level=debug msg="Calling POST /v1.41/containers/create?name=testinghp2"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.125750742Z" level=debug msg="form data: {\"AttachStderr\":true,\"AttachStdin\":false,\"AttachStdout\":true,\"Cmd\":null,\"Domainname\":\"\",\"Entrypoint\":null,\"Env\":null,\"ExposedPorts\":{\"80/tcp\":{}},\"HostConfig\":{\"AutoRemove\":false,\"Binds\":null,\"BlkioDeviceReadBps\":null,\"BlkioDeviceReadIOps\":null,\"BlkioDeviceWriteBps\":null,\"BlkioDeviceWriteIOps\":null,\"BlkioWeight\":0,\"BlkioWeightDevice\":[],\"CapAdd\":null,\"CapDrop\":null,\"Cgroup\":\"\",\"CgroupParent\":\"\",\"CgroupnsMode\":\"\",\"ConsoleSize\":[0,0],\"ContainerIDFile\":\"\",\"CpuCount\":0,\"CpuPercent\":0,\"CpuPeriod\":0,\"CpuQuota\":0,\"CpuRealtimePeriod\":0,\"CpuRealtimeRuntime\":0,\"CpuShares\":0,\"CpusetCpus\":\"\",\"CpusetMems\":\"\",\"DeviceCgroupRules\":null,\"DeviceRequests\":null,\"Devices\":[],\"Dns\":[],\"DnsOptions\":[],\"DnsSearch\":[],\"ExtraHosts\":null,\"GroupAdd\":null,\"IOMaximumBandwidth\":0,\"IOMaximumIOps\":0,\"IpcMode\":\"\",\"Isolation\":\"\",\"KernelMemory\":0,\"KernelMemoryTCP\":0,\"Links\":null,\"LogConfig\":{\"Config\":{},\"Type\":\"\"},\"MaskedPaths\":null,\"Memory\":0,\"MemoryReservation\":0,\"MemorySwap\":0,\"MemorySwappiness\":-1,\"NanoCpus\":0,\"NetworkMode\":\"default\",\"OomKillDisable\":false,\"OomScoreAdj\":0,\"PidMode\":\"\",\"PidsLimit\":0,\"PortBindings\":{\"80/tcp\":[{\"HostIp\":\"\",\"HostPort\":\"\"}]},\"Privileged\":false,\"PublishAllPorts\":false,\"ReadonlyPaths\":null,\"ReadonlyRootfs\":false,\"RestartPolicy\":{\"MaximumRetryCount\":0,\"Name\":\"no\"},\"SecurityOpt\":null,\"ShmSize\":0,\"UTSMode\":\"\",\"Ulimits\":null,\"UsernsMode\":\"\",\"VolumeDriver\":\"\",\"VolumesFrom\":null},\"Hostname\":\"\",\"Image\":\"testing123\",\"Labels\":{},\"NetworkingConfig\":{\"EndpointsConfig\":{}},\"OnBuild\":null,\"OpenStdin\":false,\"Platform\":null,\"StdinOnce\":false,\"Tty\":false,\"User\":\"\",\"Volumes\":{},\"WorkingDir\":\"\"}"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.127328464Z" level=debug msg="[zfs] ID:422ad615-c317-47f5-a44d-df7d2a543526 START zfs snapshot xpool/docker/3ce4c99ab17c53d7a927891e9592bf12bfb3038328979cbcb75fc7ce5aca372a@127313668" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.150551510Z" level=debug msg="[zfs] ID:422ad615-c317-47f5-a44d-df7d2a543526 FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.150594696Z" level=debug msg="[zfs] ID:a6748801-d0f8-4b9f-992c-f8b28a75076d START zfs list -Hp -o name,origin,used,available,mountpoint,compression,type,volsize,quota,referenced,written,logicalused,usedbydataset xpool/docker/3ce4c99ab17c53d7a927891e9592bf12bfb3038328979cbcb75fc7ce5aca372a@127313668" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.153258665Z" level=debug msg="[zfs] ID:a6748801-d0f8-4b9f-992c-f8b28a75076d FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.153291623Z" level=debug msg="[zfs] ID:9f9c3295-f5d3-45f7-b812-fe0f09cc3546 START zfs clone -p -o mountpoint=legacy xpool/docker/3ce4c99ab17c53d7a927891e9592bf12bfb3038328979cbcb75fc7ce5aca372a@127313668 xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.182345126Z" level=debug msg="[zfs] ID:9f9c3295-f5d3-45f7-b812-fe0f09cc3546 FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.182401912Z" level=debug msg="[zfs] ID:424580d0-4699-4c89-a575-b4709340984f START zfs list -Hp -o name,origin,used,available,mountpoint,compression,type,volsize,quota,referenced,written,logicalused,usedbydataset xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.185230299Z" level=debug msg="[zfs] ID:424580d0-4699-4c89-a575-b4709340984f FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.185264792Z" level=debug msg="[zfs] ID:6daef812-7dff-4fe5-a222-e4477a970676 START zfs destroy -d xpool/docker/3ce4c99ab17c53d7a927891e9592bf12bfb3038328979cbcb75fc7ce5aca372a@127313668" storage-driver=zfs
Dec 02 10:19:15 hostname zed[9766]: eid=807 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.198270236Z" level=debug msg="[zfs] ID:6daef812-7dff-4fe5-a222-e4477a970676 FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.198327965Z" level=debug msg="mount(\"xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init\", \"/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init\", \"\")" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.206877715Z" level=debug msg="unmount(\"/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init\")" storage-driver=zfs
Dec 02 10:19:15 hostname systemd[1]: var-lib-docker-zfs-graph-d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\x2dinit.mount: Succeeded.
Dec 02 10:19:15 hostname systemd[5380]: var-lib-docker-zfs-graph-d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\x2dinit.mount: Succeeded.
Dec 02 10:19:15 hostname systemd[3391]: var-lib-docker-zfs-graph-d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\x2dinit.mount: Succeeded.
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.237048312Z" level=debug msg="[zfs] ID:7d006a89-0c57-4046-ab19-e75a372c9a14 START zfs snapshot xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init@237018049" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.255736266Z" level=debug msg="[zfs] ID:7d006a89-0c57-4046-ab19-e75a372c9a14 FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.255779211Z" level=debug msg="[zfs] ID:4e95a06b-aee9-4931-b290-23fdb82bfe12 START zfs list -Hp -o name,origin,used,available,mountpoint,compression,type,volsize,quota,referenced,written,logicalused,usedbydataset xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init@237018049" storage-driver=zfs
Dec 02 10:19:15 hostname zed[9906]: eid=808 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.258352496Z" level=debug msg="[zfs] ID:4e95a06b-aee9-4931-b290-23fdb82bfe12 FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.258388079Z" level=debug msg="[zfs] ID:7d27af7b-2164-4166-b07f-0a772d8d1b4f START zfs clone -p -o mountpoint=legacy xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init@237018049 xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.287705082Z" level=debug msg="[zfs] ID:7d27af7b-2164-4166-b07f-0a772d8d1b4f FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.287760334Z" level=debug msg="[zfs] ID:ec68fada-064f-4167-9573-5469c804a5dc START zfs list -Hp -o name,origin,used,available,mountpoint,compression,type,volsize,quota,referenced,written,logicalused,usedbydataset xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.290567068Z" level=debug msg="[zfs] ID:ec68fada-064f-4167-9573-5469c804a5dc FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.290601070Z" level=debug msg="[zfs] ID:18cd3f27-42ad-492c-bf31-80e4fc18ccbe START zfs destroy -d xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e-init@237018049" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.304338991Z" level=debug msg="[zfs] ID:18cd3f27-42ad-492c-bf31-80e4fc18ccbe FINISH" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.307544621Z" level=debug msg="mount(\"xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\", \"/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\", \"\")" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.310489285Z" level=debug msg="container mounted via layerStore: &{/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e 0x561ff5189f60 0x561ff5189f60}" container=f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.310549286Z" level=debug msg="unmount(\"/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\")" storage-driver=zfs
Dec 02 10:19:15 hostname zed[10007]: eid=809 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname systemd[5380]: var-lib-docker-zfs-graph-d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e.mount: Succeeded.
Dec 02 10:19:15 hostname systemd[3391]: var-lib-docker-zfs-graph-d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e.mount: Succeeded.
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.345475795Z" level=debug msg="Calling POST /v1.41/containers/f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220/attach?stderr=1&stdout=1&stream=1"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.345561293Z" level=debug msg="attach: stdout: begin"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.345564439Z" level=debug msg="attach: stderr: begin"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.345764796Z" level=debug msg="Calling POST /v1.41/containers/f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220/wait?condition=next-exit"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.346002584Z" level=debug msg="Calling POST /v1.41/containers/f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220/start"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.346440314Z" level=debug msg="mount(\"xpool/docker/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\", \"/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e\", \"\")" storage-driver=zfs
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.348211874Z" level=debug msg="container mounted via layerStore: &{/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e 0x561ff5189f60 0x561ff5189f60}" container=f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.348526456Z" level=debug msg="Assigning addresses for endpoint testinghp2's interface on network bridge"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.348544942Z" level=debug msg="RequestAddress(LocalDefault/172.17.0.0/16, <nil>, map[])"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.348564508Z" level=debug msg="Request address PoolID:172.17.0.0/16 App: ipam/default/data, ID: LocalDefault/172.17.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65532, Sequence: (0xe0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:4 Serial:false PrefAddress:<nil> "
Dec 02 10:19:15 hostname kernel: docker0: port 2(veth67a9668) entered blocking state
Dec 02 10:19:15 hostname kernel: docker0: port 2(veth67a9668) entered disabled state
Dec 02 10:19:15 hostname kernel: device veth67a9668 entered promiscuous mode
Dec 02 10:19:15 hostname networkd-dispatcher[2004]: WARNING:Unknown index 59 seen, reloading interface list
Dec 02 10:19:15 hostname systemd-networkd[1932]: veth67a9668: Link UP
Dec 02 10:19:15 hostname systemd-udevd[9788]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
Dec 02 10:19:15 hostname systemd-udevd[9788]: Using default interface naming scheme 'v245'.
Dec 02 10:19:15 hostname systemd-udevd[9788]: veth2215ea0: Could not generate persistent MAC: No data available
Dec 02 10:19:15 hostname systemd-udevd[10073]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
Dec 02 10:19:15 hostname systemd-udevd[10073]: Using default interface naming scheme 'v245'.
Dec 02 10:19:15 hostname systemd-udevd[10073]: veth67a9668: Could not generate persistent MAC: No data available
Dec 02 10:19:15 hostname NetworkManager[1956]: <info>  [1638440355.3588] manager: (veth2215ea0): new Veth device (/org/freedesktop/NetworkManager/Devices/80)
Dec 02 10:19:15 hostname NetworkManager[1956]: <info>  [1638440355.3602] manager: (veth67a9668): new Veth device (/org/freedesktop/NetworkManager/Devices/81)
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.362925189Z" level=debug msg="Assigning addresses for endpoint testinghp2's interface on network bridge"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.373754396Z" level=debug msg="Programming external connectivity on endpoint testinghp2 (4bcacfa7aa21acb351350fe72cd775201e86a9fcc50254ed78e98c9e64f28efe)"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.373841749Z" level=debug msg="/usr/sbin/iptables, [--wait -t nat -C DOCKER -p tcp -d 0/0 --dport 49155 -j DNAT --to-destination 172.17.0.3:80 ! -i docker0]"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.375207931Z" level=debug msg="/usr/sbin/iptables, [--wait -t nat -A DOCKER -p tcp -d 0/0 --dport 49155 -j DNAT --to-destination 172.17.0.3:80 ! -i docker0]"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.376479254Z" level=debug msg="/usr/sbin/iptables, [--wait -t filter -C DOCKER ! -i docker0 -o docker0 -p tcp -d 172.17.0.3 --dport 80 -j ACCEPT]"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.377514165Z" level=debug msg="/usr/sbin/iptables, [--wait -t filter -A DOCKER ! -i docker0 -o docker0 -p tcp -d 172.17.0.3 --dport 80 -j ACCEPT]"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.378596136Z" level=debug msg="/usr/sbin/iptables, [--wait -t nat -C POSTROUTING -p tcp -s 172.17.0.3 -d 172.17.0.3 --dport 80 -j MASQUERADE]"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.379576124Z" level=debug msg="/usr/sbin/iptables, [--wait -t nat -A POSTROUTING -p tcp -s 172.17.0.3 -d 172.17.0.3 --dport 80 -j MASQUERADE]"
Dec 02 10:19:15 hostname zed[10086]: eid=810 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.384706822Z" level=debug msg="EnableService f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220 START"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.384724986Z" level=debug msg="EnableService f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220 DONE"
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.392243987Z" level=debug msg="bundle dir created" bundle=/var/run/docker/containerd/f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220 module=libcontainerd namespace=moby root=/var/lib/docker/zfs/graph/d2a1523d2c44ef6b3f131b29a924a90cd82f44e3b178bfcf6949e3ceff91626e
Dec 02 10:19:15 hostname containerd[792496]: time="2021-12-02T10:19:15.402598347Z" level=info msg="starting signal loop" namespace=moby path=/run/containerd/io.containerd.runtime.v2.task/moby/f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220 pid=10107
Dec 02 10:19:15 hostname zed[10128]: eid=811 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname systemd[3391]: run-docker-runtime\x2drunc-moby-f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220-runc.BrLXkQ.mount: Succeeded.
Dec 02 10:19:15 hostname systemd[5380]: run-docker-runtime\x2drunc-moby-f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220-runc.BrLXkQ.mount: Succeeded.
Dec 02 10:19:15 hostname systemd[1]: run-docker-runtime\x2drunc-moby-f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220-runc.BrLXkQ.mount: Succeeded.
Dec 02 10:19:15 hostname zed[10146]: eid=812 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname zed[10159]: eid=813 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname kernel: eth0: renamed from veth2215ea0
Dec 02 10:19:15 hostname zed[10162]: eid=814 class=history_event pool_guid=0x6177DCB6661C6C50
Dec 02 10:19:15 hostname systemd-networkd[1932]: veth67a9668: Gained carrier
Dec 02 10:19:15 hostname NetworkManager[1956]: <info>  [1638440355.6083] device (veth67a9668): carrier: link connected
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.609813280Z" level=debug msg="sandbox set key processing took 113.281059ms for container f592ad687f24024e2cb91e95cd3fd017a181182072acb97ada18a86623203220"
Dec 02 10:19:15 hostname kernel: IPv6: ADDRCONF(NETDEV_CHANGE): veth67a9668: link becomes ready
Dec 02 10:19:15 hostname kernel: docker0: port 2(veth67a9668) entered blocking state
Dec 02 10:19:15 hostname kernel: docker0: port 2(veth67a9668) entered forwarding state
Dec 02 10:19:15 hostname gnome-shell[5932]: Removing a network device that was not added
Dec 02 10:19:15 hostname gnome-shell[4946]: Removing a network device that was not added
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.632982773Z" level=debug msg=event module=libcontainerd namespace=moby topic=/tasks/create
Dec 02 10:19:15 hostname dockerd[4187675]: time="2021-12-02T10:19:15.663396482Z" level=debug msg=event module=libcontainerd namespace=moby topic=/tasks/start
```
I could've grepped out the zfs stuff but I wanted to provide it unmodified. Looks like the commit you referenced could cause this, but I am not familiar enough to say I am afraid.

**Comment by thaJeztah** [comment]
> I do not see any message similar to `falling back to default port range...`, even in debug mode.

That's a pity (would've made debugging easier 😅)

Thanks for checking, and for the extra info!

**Comment by jamiegDS** [comment]
Hi,

Is there any chance of some action here? Either, the fix could be merged, or we drop it, update the docs and going forward the ephemeral range is 49153-65535. I fear the longer this stays here, the more likely someone else starts relying on it being  49153-65535, which would then cause them problems if this fix is applied.

Thanks, appreciate it!

**Comment by reid-harrison** [comment]
This fix is released with [v22.06 beta](https://github.com/moby/moby/releases/tag/v22.06.0-beta.0). Is there any intention to fix in 20.10?
